src/release.rs

```diff
@@ -306,3 +404,113 @@ pub fn produce_install_only(tar_zst_path: &Path) -> Result<PathBuf> {
+        .expect("should have filename")
+        .to_string_lossy();
+
+    let mut name_parts = filename
+        .split('-')
+        .map(|x| x.to_string())
+        .collect::<Vec<_>>();
+    let parts_len = name_parts.len();
+
+    name_parts[parts_len - 1] = "install_only_stripped".to_string();
+
+    let install_only_name = name_parts.join("-");
+    let install_only_name = format!("{install_only_name}.tar.gz");
+
+    let dest_path = tar_gz_path.with_file_name(install_only_name);
+    std::fs::write(&dest_path, gz_data)?;
+
+    Ok(dest_path)
+}
+
+/// URL from which to download LLVM.
+///
+/// To be kept in sync with `pythonbuild/downloads.py`.
+static LLVM_URL: Lazy<Url> = Lazy::new(|| {
+    if cfg!(target_os = "macos") {
+        if std::env::consts::ARCH == "aarch64" {
+            Url::parse("https://github.com/indygreg/toolchain-tools/releases/download/toolchain-bootstrap%2F20240713/llvm-18.0.8+20240713-aarch64-apple-darwin.tar.zst").unwrap()
+        } else if std::env::consts::ARCH == "x86_64" {
+            Url::parse("https://github.com/indygreg/toolchain-tools/releases/download/toolchain-bootstrap%2F20240713/llvm-18.0.8+20240713-x86_64-apple-darwin.tar.zst").unwrap()
+        } else {
+            panic!("unsupported macOS architecture");
+        }
+    } else if cfg!(target_os = "linux") {
+        Url::parse("https://github.com/indygreg/toolchain-tools/releases/download/toolchain-bootstrap%2F20240713/llvm-18.0.8+20240713-gnu_only-x86_64-unknown-linux-gnu.tar.zst").unwrap()
+    } else {
+        panic!("unsupported platform");
+    }
+});
+
+/// Bootstrap `llvm` for the current platform.
+///
+/// Returns the path to the top-level `llvm` directory.
+pub async fn bootstrap_llvm() -> Result<PathBuf> {
+    let url = &*LLVM_URL;
+    let filename = url.path_segments().unwrap().last().unwrap();
+
+    let llvm_dir = PathBuf::from("llvm");
+
+    // If `llvm` is already available with the target version, return it.
+    if llvm_dir.join(filename).exists() {
+        return Ok(llvm_dir.join("llvm"));
+    }
+
+    println!("Downloading LLVM tarball from: {url}");
+
+    // Create a temporary directory to download and extract the LLVM tarball.
+    let temp_dir = tempfile::TempDir::new()?;
+
+    // Download the tarball.
+    let tarball_path = temp_dir
+        .path()
+        .join(url.path_segments().unwrap().last().unwrap());
+    let mut tarball_file = tokio::fs::File::create(&tarball_path).await?;
+    let mut bytes_stream = reqwest::Client::new()
+        .get(url.clone())
+        .send()
+        .await?
+        .bytes_stream();
+    while let Some(chunk) = bytes_stream.next().await {
+        tokio::io::copy(&mut chunk?.as_ref(), &mut tarball_file).await?;
+    }
+
+    // Decompress the tarball.
+    let tarball = std::fs::File::open(&tarball_path)?;
+    let tar = zstd::stream::Decoder::new(std::io::BufReader::new(tarball))?;
+    let mut archive = tar::Archive::new(tar);
+    archive.unpack(temp_dir.path())?;
```

Comment:
There might be a way to do this without having to buffer the entire file first. But it's probably not worth the optimization.